lib/find-visualstudio.js

```diff
@@ -128,24 +175,35 @@ class VisualStudioFinder {
     ]
 
     this.log.silly('Running', ps, psArgs)
-    const [err, stdout, stderr] = await execFile(ps, psArgs, { encoding: 'utf8' })
-    return this.parseData(err, stdout, stderr)
+    const [err, stdout, stderr] = await this.execFile(ps, psArgs)
+    const parsedData = this.parseData(err, stdout, stderr, { checkIsArray: true })
+    if (parsedData === null) {
+      return null
+    }
+    return this.processData(parsedData)
   }
 
-  // Parse the output of the PowerShell script and look for an installation
-  // of Visual Studio 2017 or newer to use
-  parseData (err, stdout, stderr) {
+  // Parse the output of the PowerShell script, make sanity checks
+  parseData (err, stdout, stderr, sanityCheckOptions) {
+    const defaultOptions = {
+      checkIsArray: false
+    }
+
+    // Merging provided options with the default options
+    const sanityOptions = { ...defaultOptions, ...sanityCheckOptions }
+
     this.log.silly('PS stderr = %j', stderr)
 
-    const failPowershell = () => {
+    const failPowershell = (failureDetails) => {
       this.addLog(
-        'could not use PowerShell to find Visual Studio 2017 or newer, try re-running with \'--loglevel silly\' for more details')
+        `could not use PowerShell to find Visual Studio 2017 or newer, try re-running with '--loglevel silly' for more details. \n
+        Failure details: ${failureDetails}`)
       return null
     }
 
     if (err) {
       this.log.silly('PS err = %j', err && (err.stack || err))
-      return failPowershell()
+      return failPowershell(`${err}`.substring(0, 40))
```

Comment:
Just out of curiosity, why cut at 40?

Reply:
Just to give a possible hint where the failure is and then look for more details in the log or `--verbose` output.
Didn't want to put too large number which can clog the output. But it can be something around 70-90 too (usual line length)